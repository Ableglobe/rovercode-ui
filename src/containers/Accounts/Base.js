import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Grid, Image } from 'semantic-ui-react';
import PropTypes from 'prop-types';

import NotFound from '@/containers/Global/NotFound';
import logoImage from '@/assets/images/rovercode_logo.png';

import Login from './Login';
import LoginCallback from './LoginCallback';
import PasswordReset from './PasswordReset';
import PasswordResetCallback from './PasswordResetCallback';
import SignUp from './SignUp';


const Base = ({ match }) => (
  <Grid centered columns={16}>
    <Grid.Row>
      <Image src={logoImage} />
    </Grid.Row>
    <Grid.Row>
      <Grid.Column width={16}>
        <Switch>
          <Route exact path={`${match.path}/login`} component={Login} />
          <Route exact path={`${match.path}/login/callback/:service`} component={LoginCallback} />
          <Route exact path={`${match.path}/reset`} component={PasswordReset} />
          <Route exact path={`${match.path}/reset/callback/:uid/:token`} component={PasswordResetCallback} />
          <Route exact path={`${match.path}/signup`} component={SignUp} />
          <Route component={NotFound} />
        </Switch>
      </Grid.Column>
    </Grid.Row>
  </Grid>
);

Base.propTypes = {
  match: PropTypes.shape({
    path: PropTypes.string.isRequired,
  }).isRequired,
};

export default Base;
