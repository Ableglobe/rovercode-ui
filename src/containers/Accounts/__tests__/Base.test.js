import React from 'react';
import { Route } from 'react-router-dom';
import { Image } from 'semantic-ui-react';
import { shallow } from 'enzyme';

import Base from '../Base';

test('Base renders on the page with no errors', () => {
  const match = {
    path: '/accounts',
  };

  const wrapper = shallow(<Base match={match} />);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find(Image).exists()).toBe(true);
  expect(wrapper.find(Route).length).toBe(2);
  expect(wrapper.find(Route).at(0).prop('path')).toBe('/accounts/login');
  expect(wrapper.find(Route).at(1).prop('path')).toBe('/accounts/signup');
});
