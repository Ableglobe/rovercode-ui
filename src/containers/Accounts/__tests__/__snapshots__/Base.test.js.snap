// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base renders on the page with no errors 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Base
    match={
      Object {
        "path": "/accounts",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "centered": true,
      "children": Array [
        <GridRow>
          <Image
            as="img"
            src={Object {}}
            ui={true}
          />
        </GridRow>,
        <GridRow>
          <GridColumn
            width={16}
          >
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/login/callback/:service"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/reset"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/reset/callback/:uid/:token"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/signup"
              />
              <Route
                component={[Function]}
              />
            </Switch>
          </GridColumn>
        </GridRow>,
      ],
      "columns": 16,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Image
            as="img"
            src={Object {}}
            ui={true}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "img",
            "src": Object {},
            "ui": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GridColumn
            width={16}
          >
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/login/callback/:service"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/reset"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/reset/callback/:uid/:token"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/signup"
              />
              <Route
                component={[Function]}
              />
            </Switch>
          </GridColumn>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/login/callback/:service"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/reset"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/reset/callback/:uid/:token"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/accounts/signup"
              />
              <Route
                component={[Function]}
              />
            </Switch>,
            "width": 16,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/login"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/login/callback/:service"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/reset"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/reset/callback/:uid/:token"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/signup"
                />,
                <Route
                  component={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/accounts/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/accounts/login/callback/:service",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/accounts/reset",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/accounts/reset/callback/:uid/:token",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/accounts/signup",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "centered": true,
        "children": Array [
          <GridRow>
            <Image
              as="img"
              src={Object {}}
              ui={true}
            />
          </GridRow>,
          <GridRow>
            <GridColumn
              width={16}
            >
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/login"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/login/callback/:service"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/reset"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/reset/callback/:uid/:token"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/signup"
                />
                <Route
                  component={[Function]}
                />
              </Switch>
            </GridColumn>
          </GridRow>,
        ],
        "columns": 16,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Image
              as="img"
              src={Object {}}
              ui={true}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "img",
              "src": Object {},
              "ui": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GridColumn
              width={16}
            >
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/login"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/login/callback/:service"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/reset"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/reset/callback/:uid/:token"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/signup"
                />
                <Route
                  component={[Function]}
                />
              </Switch>
            </GridColumn>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/login"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/login/callback/:service"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/reset"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/reset/callback/:uid/:token"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/accounts/signup"
                />
                <Route
                  component={[Function]}
                />
              </Switch>,
              "width": 16,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/accounts/login"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/accounts/login/callback/:service"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/accounts/reset"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/accounts/reset/callback/:uid/:token"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/accounts/signup"
                  />,
                  <Route
                    component={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/accounts/login",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/accounts/login/callback/:service",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/accounts/reset",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/accounts/reset/callback/:uid/:token",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/accounts/signup",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
